app/src/main/java/com/example/geoquiz/CheatActivity.kt

```diff
@@ -0,0 +1,56 @@
+package com.example.geoquiz
+
+import android.app.Activity
+import android.content.Context
+import android.content.Intent
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.widget.Button
+import android.widget.TextView
+
+private const val EXTRA_ANSWER_IS_TRUE = "com.example.geoquiz.answer_is_true"
+const val EXTRA_ANSWER_SHOWN = "com.example.geoquiz.answer_shown"
+
+class CheatActivity : AppCompatActivity() {
+
+    private lateinit var answerTextView: TextView
+    private lateinit var showAnswerButton: Button
+
+    private var answerIsTrue = false
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_cheat)
+
+        answerIsTrue = intent.getBooleanExtra(EXTRA_ANSWER_IS_TRUE, false)
+
+        answerTextView = findViewById(R.id.answer_text_view)
+        showAnswerButton = findViewById(R.id.show_answer_button)
+
+        showAnswerButton.setOnClickListener {
+            val answerText = when {
```

Comment:
I would set the answerTextView text always and make it's visibility invisible. Then show it when the button is clickked. The code is easier to read that way.
Afaik (as far as I know), there is no way of reading invisible fields such as on the web with inspect element